src/client/caching/structures/guilds/GuildMember.ts

```diff
@@ -74,14 +74,16 @@ export class GuildMember extends Structure {
 		return this.nick ? `<@!${this.id}>` : `<@${this.id}>`;
 	}
 
-	protected _patch(data: APIGuildMemberData): this {
-		this.deaf = data.deaf;
-		this.joinedTimestamp = new Date(data.joined_at).getTime();
-		this.mute = data.mute;
-		this.nick = data.nick;
+	protected _patch(data: MemberData): this {
+		this.deaf = 'deaf' in data ? data.deaf : null;
+		this.joinedTimestamp = 'joined_at' in data ? new Date(data.joined_at).getTime() : null;
+		this.mute = 'mute' in data ? data.mute : null;
+		this.nick = 'nick' in data ? data.nick : null;
```

Comment:
Let's make a consistent use of checking props, let's use just Reflect.has please!
```suggestion
		this.deaf = Reflect.has(data, 'deaf') ? data.deaf : null;
		this.joinedTimestamp = Reflect.has(data, 'joined_at') ? new Date(data.joined_at).getTime() : null;
		this.mute = Reflect.has(data, 'mute') ? data.mute : null;
		this.nick = Reflect.has(data, 'nick') ? data.nick : null;
```

Reply:
Can't be done, Reflect.has doesn't tell TSC that the property exists, unlike the `in` operator.

And no, casting doesn't work.

Reply:
☹️